src/main/kotlin/me/partlysanestudios/partlysaneskies/features/debug/DebugKey.kt

```diff
@@ -102,4 +115,48 @@ object DebugKey {
             event.pipeline.add(Waypoint("Debug Waypoint", waypointPoint.toBlockPosInt()))
         }
     }
+
+    private var cylinderPoint = Point3d(0.0, 0.0, 0.0)
+    @SubscribeEvent
+    fun onWorldRenderLast(event: RenderWorldLastEvent) {
+
+        if (!(isDebugMode() && config.debugCylinder)) {
+            return
+        }
+        //            Sets the correct state
+        val renderManager = PartlySaneSkies.minecraft.renderManager
+
+        GlStateManager.pushMatrix()
+        GlStateManager.translate(-renderManager.viewerPosX, -renderManager.viewerPosY, -renderManager.viewerPosZ)
+
+        GlStateManager.disableTexture2D()
+        GlStateManager.disableCull()
+        GlStateManager.enableBlend()
+        GlStateManager.disableLighting()
+
+        GlStateManager.tryBlendFuncSeparate(GL11.GL_SRC_ALPHA, GL11.GL_ONE_MINUS_SRC_ALPHA, GL11.GL_ONE, GL11.GL_ZERO)
+        GL11.glLineWidth(4.0f)
+
+
+//            Gets the tessellator
+        val tessellator = Tessellator.getInstance() // from my understanding it's just a tesseract but for nerdier nerds
+        val worldRenderer = tessellator.worldRenderer
+
+        worldRenderer.begin(GL11.GL_QUADS, DefaultVertexFormats.POSITION)
+        GlStateManager.color(ThemeManager.accentColor.toJavaColor().red/255f, ThemeManager.accentColor.toJavaColor().green/255f, ThemeManager.accentColor.toJavaColor().blue/255f, (ThemeManager.accentColor.toJavaColor().alpha/255f) * .667f)
+        worldRenderer.drawCylinderFill(cylinderPoint, 8.0, 20.0)
```

Comment:
Why do we have to call all this weird stuff before being able to render it, feels weird when wanting to use it

Reply:
Why dont you add a main rendering function that does all of this + render this euclid

Reply:
Because it depends on the settings we want
For example, the waypoint code is different than this on, which his different than the range highlight

Reply:
But like why not just add a RenderEuclid.render() that 
1. calls all of that glstatemanager shit
2. calls worldrenderer.render...
3. calls all of that glstatemanager shit again

this will make it a lot easier to use

Reply:
This isn't supposed to plop a circle in the world. It's supposed to be like everything in RenderPrimatives, where they can then be used to render things in the world. (Example: look at the implementation of waypoints)

Reply:
This is just abstracting the logic and maths to be used for more complex things

Reply:
Rendering a cube in renderprimatives would require the same things. This is just math to draw the points needed for circle into the tesslator buffer. When I do inevitably use it, then I will have a .renderCylinder function or whatever, but that's not for now

Reply:
Yeah okay